### PR TITLE
Move config into flyer lsdc 133

### PR DESCRIPTION
**Pull request by JunAishima**
Code to incorporate all of the main setup functions (flyer functions setup_eiger_arming() and actual_scan()) into the flyer object. Note the use of `object.put(value)` instead of using `bps.mv(object, value)`

The corresponding functions have been removed from scans.py as they are no longer necessary.

https://github.com/NSLS-II/lsdc/issues/133
https://github.com/NSLS-II/lsdc/issues/137

### CODE REVIEW

File: mxtools/flyer.py [hunk 1]
```diff
@@ -167,98 +167,147 @@ def unstage(self):
...
+    def setup_vector_program(self, num_images, angle_start, angle_end, exposure_period_per_image):
+        self.vector.num_frames.put(num_images)
+        self.vector.start.omega.put(angle_start)
+        self.vector.end.omega.put(angle_end)
+        self.vector.frame_exptime.put(exposure_period_per_image * 1000.0)
+        self.vector.hold.put(0)
+
+    def zebra_daq_prep(self):
+        self.zebra.reset.put(1)
+        ttime.sleep(2.0)
```
Comment (RobertSchaffer1) [inline mxtools/flyer.py:286]: We don't need to do this asynchronously (ie. yield from)?

File: mxtools/flyer.py [hunk 2]
```diff
@@ -167,98 +167,147 @@ def unstage(self):
...
+        angle_start = kwargs["angle_start"]
+        exposurePeriodPerImage = kwargs["exposurePeriodPerImage"]
+        detector_dead_time = kwargs["detector_dead_time"]
+        scanWidth = kwargs["scanWidth"]
+        imgWidth = kwargs["imgWidth"]
+        numImages = kwargs["numImages"]
+        self.zebra_daq_prep()
+        ttime.sleep(1.0)
 
-def configure_flyer(
```
Comment (RobertSchaffer1) [inline mxtools/flyer.py:171]: What is the new/old workflow when using these? You would have had to call actual_scan() from somewhere, presumably? Now you have to call update_parameters from somewhere?

File: mxtools/flyer.py (hunk 1 above)
Comment (JunAishima) [reply mxtools/flyer.py:286]: nope, Maksim recommended that we do it this way. this and the following question are answered by looking at the pseudocode show in issue 137: https://github.com/NSLS-II/lsdc/issues/137 
`flyer.update_parameters(**kwargs)`
`RE(bps.fly([flyer])`

These calls would need to be done in either test code or within LSDC itself.

File: mxtools/flyer.py (hunk 2 above)
Comment (JunAishima) [reply mxtools/flyer.py:171]: as shown in the pseudocode above, we have replaced all of the code that is used by the arming function and actual_scan and moved it into flyer, so you just update_parameters(), then fly

**Review by RobertSchaffer1 — APPROVED** [review]

**Comment by RobertSchaffer1** [comment]
Approved subject to successful testing, the code looks good but there are enough moving pieces I can't swear it will all work.

**Comment by JunAishima** [comment]
> Approved subject to successful testing, the code looks good but there are enough moving pieces I can't swear it will all work.

Neither can I, though I did test it yesterday and the collection started - so vector must have been reasonably correct, zebra correct, and Eiger setup reasonably correct as well. will be checking through the dataset metadata to check deeper.

**Pull request merged**